src/class/hid/hid_device.h

```diff
@@ -118,6 +118,8 @@ TU_ATTR_WEAK bool tud_hid_set_idle_cb(uint8_t instance, uint8_t idle_rate);
 // Note: For composite reports, report[0] is report ID
 TU_ATTR_WEAK void tud_hid_report_complete_cb(uint8_t instance, uint8_t const* report, uint16_t len);
 
+// Invoked when received SOF interrupt
+TU_ATTR_WEAK void tud_hid_sof_cb(uint32_t frame_count);
```

Comment:
As a callback it should be invoked in the ISR due to the SOF event and called here: https://github.com/hathach/tinyusb/blob/cfe8ca3fa51061bbe7e519b72236f9aabae13e59/src/device/usbd.c#L599